api.d.ts

```diff
@@ -1113,8 +1113,56 @@ export interface ILexerConfig {
      * Note that this would have negative performance implications.
      */
     safeMode?: boolean
+
+    /**
+     * A custom error message provider.
+     * Can be used to override the default error messages.
+     * For example:
+     *   - Translating the error messages to a different languages.
+     *   - Changing the formatting.
+     */
+    errorMessageProvider?: ILexerErrorMessageProvider
 }
 
+export interface ILexerErrorMessageProvider {
+    /**
+     * Unexpected Character Error happens when lexer could not find token match for input text
+     *
+     * @param text - Input text.
+     *
+     * @param errorStartOffset - Offset in input text where error starts.
+     *
+     * @param errorLength - Error length.
+     *
+     * @param errorLine - Line number where the error occured.
+     *
+     * @param errorColumn - Column number where the error occured.
+     */
+    buildUnexpectedCharacterMessage(
+        text: string,
+        errorStartOffset: number,
+        errorLength: number,
```

Comment:
all these arguments are prefixed by "error" in their name.
Does this serve a purpose?
- Look for "Smurf" here: https://blog.codinghorror.com/new-programming-jargon/